scanpointgenerator/generators/lissajousgenerator.py

```diff
@@ -54,25 +59,25 @@ def __init__(self, names, units, box, num_lobes, num_points=None):
 
         self.axes = self.names  # For GDA
 
-    def _calc(self, i):
-        """Calculate the coordinate for a given index"""
-        x = self.centre[0] + \
-            self.x_max * m.sin(self.x_freq * i * self.increment +
-                               self.phase_diff)
-        y = self.centre[1] + \
-            self.y_max * m.sin(self.y_freq * i * self.increment)
-
-        return x, y
-
-    def iterator(self):
-        for i in range_(self.num):
-            p = Point()
-            p.positions[self.names[0]], p.positions[self.names[1]] = self._calc(i)
-            p.lower[self.names[0]], p.lower[self.names[1]] = self._calc(i - 0.5)
-            p.upper[self.names[0]], p.upper[self.names[1]] = self._calc(i + 0.5)
-            p.indexes = [i]
-
-            yield p
+    def produce_points(self):
+        self.points = {}
+        self.bounds = {}
+
+        x0, y0 = self.centre[0], self.centre[1]
+        A, B = self.x_max, self.y_max
+        a, b = self.x_freq, self.y_freq
+        d = self.phase_diff
+        fx = lambda t: x0 + A * np.sin(a * 2*m.pi * t/self.num + d)
+        fy = lambda t: y0 + B * np.sin(b * 2*m.pi * t/self.num)
+        x = fx(np.arange(self.num))
```

Comment:
Now you've put it like this, can we push some of this to the base class? We could have produce_points() take an array of indexes (either np.arange(self.num) or np.arange(self.num + 1) - 0.5) and return the points or the bounds array. This would remove duplication in the Generators, which is good because we may want to write lots of them...